src/java.base/share/classes/java/util/ResourceBundle.java

```diff
@@ -1546,7 +1584,23 @@ private static ResourceBundle getBundleImpl(String baseName,
                                                 Locale locale,
                                                 Class<?> caller,
                                                 Control control) {
-        return getBundleImpl(baseName, locale, caller, caller.getClassLoader(), control);
+        final ClassLoader loader = (caller != null) ?
+                caller.getClassLoader() : getLoader(getCallerModule(caller));
```

Comment:
```suggestion
              ClassLoader loader = getLoader(getCallerModule(caller));
```

This can be simplified as above.  I would drop `final` too as it only adds noise to the code.